src/command.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Command implementations. Interacts with parsec-client-rust.
+
+use crate::cli::{Action, CommandLineContext};
+use crate::error::ParsecCliError;
+use anyhow::{Context, Result};
+use parsec_client::auth::AuthenticationData;
+use parsec_client::core::interface::requests::ProviderID;
+use parsec_client::BasicClient;
+
+fn get_unauthenticated_client() -> BasicClient {
+    BasicClient::new(AuthenticationData::None)
+}
+
+/// Gets the list of availble opcodes for a given provider and outputs them.
+fn list_opcodes(
+    provider: &str,
+    _command_line_context: &CommandLineContext,
```

Comment:
I'd say, for all these functions, if you don't need this argument you shouldn't really pass it in/request it

Reply:
Sure, will remove.

Reply:
Actually, I'm not so sure about this -- in the future it's likely that we'll want to show more/less information depending on the verbosity, which is encoded in the command line context. I think this is a good thing to keep around, even if we're not using it right now.

Reply:
Makes sense